main.go

```diff
@@ -0,0 +1,128 @@
+	flagLogLevel                 = flag.String("log-level", getEnv("LOG_LEVEL", "info"), "Log level")
+	flagRemoteAPIURL             = flag.String("remote-api-url", getEnv("REMOTE_API_URL", ""), "Remote Kubernetes API server URL")
+	flagRemoteCAURL              = flag.String("remote-ca-url", getEnv("REMOTE_CA_URL", ""), "Remote Kubernetes CA certificate URL")
+	flagRemoteSATokenPath        = flag.String("remote-sa-token-path", getEnv("REMOTE_SERVICE_ACCOUNT_TOKEN_PATH", ""), "Remote Kubernetes cluster token path")
+	flagResyncPeriod             = flag.Duration("resync-period", 60*time.Minute, "Pod watcher cache resync period")
+	flagSetsPrefix               = flag.String("sets-prefix", getEnv("SETS_PREFIX", ""), "(required) A prefix used when creating network sets, needed in case of multiple sync instances for different clusters.")
+
+	saToken  = os.Getenv("WS_REMOTE_SERVICE_ACCOUNT_TOKEN")
+	bearerRe = regexp.MustCompile(`[A-Z|a-z0-9\-\._~\+\/]+=*`)
+)
+
+func usage() {
+	flag.Usage()
+	os.Exit(1)
+}
+
+func getEnv(key, defaultValue string) string {
+	value := os.Getenv(key)
+	if len(value) == 0 {
+		return defaultValue
+	}
+	return value
+}
+
+func main() {
+	flag.Parse()
+	log.InitLogger("kube-policy-semaphore", *flagLogLevel)
+	if *flagSetsPrefix == "" {
+		log.Logger.Error("Must specify non-empty prefix for the created globalnetworksets")
+		usage()
+	}
+	if *flagRemoteSATokenPath != "" {
+		data, err := ioutil.ReadFile(*flagRemoteSATokenPath)
+		if err != nil {
+			fmt.Printf("Cannot read file: %s", *flagRemoteSATokenPath)
+			os.Exit(1)
+		}
+		saToken = string(data)
+	}
+
+	if saToken != "" {
+		saToken = strings.TrimSuffix(saToken, "\n")
+		if !bearerRe.Match([]byte(saToken)) {
+			log.Logger.Error(
+				"The provided token does not match regex",
+				"regex", bearerRe.String)
+			os.Exit(1)
+		}
+	}
+
+	homeCalicoClient, err := calico.NewClient(*flagKubeConfigPath)
+	if err != nil {
+		log.Logger.Error(
+			"cannot create kube client for homecluster",
+			"err", err,
+		)
+		usage()
+	}
+	var remoteClient *kubernetes.Clientset
+	if *flagTargetKubeConfigPath != "" {
+		remoteClient, err = kube.ClientFromConfig(*flagTargetKubeConfigPath)
+	} else {
+		remoteClient, err = kube.Client(saToken, *flagRemoteAPIURL, *flagRemoteCAURL)
+	}
+	if err != nil {
+		log.Logger.Error(
+			"cannot create kube client for remotecluster",
+			"err", err,
+		)
+		usage()
+	}
+
+	r := newRunner(
+		homeCalicoClient,
+		remoteClient,
+		*flagSetsPrefix,
+		*flagLabelSelector,
+		*flagNetworkSetNameAnnotation,
+		*flagResyncPeriod,
+	)
+	if err := r.Run(); err != nil {
+		log.Logger.Error("Failed to start runner", "err", err)
+		os.Exit(1)
+	}
+
+	sm := http.NewServeMux()
+	sm.HandleFunc("/healthz", func(w http.ResponseWriter, _ *http.Request) {
+		if r.Healthy() {
+			w.WriteHeader(http.StatusOK)
+		} else {
+			w.WriteHeader(http.StatusServiceUnavailable)
+		}
+	})
+	log.Logger.Error(
+		"Listen and Serve",
+		"err", http.ListenAndServe(":8080", sm),
```

Comment:
This is not always Error level. You get an error when the server is shut down gracefully. Although practically it's not important since the process exits afterwards.

Reply:
This is inspired from here: https://golang.org/pkg/net/http/#ListenAndServe
ListenAndServe returns an error type and is treated as a fatal one there since you stop execution afterwards. I think that you need to see at least `ERROR` on your logs since this will cause your app to exit.

Reply:
What I meant was:

>  ErrServerClosed is returned by the Server's Serve, ServeTLS, ListenAndServe, and ListenAndServeTLS methods after a call to Shutdown or Close. 

Which is not really an Error, if you've called Shutdown or Close :)